data/user.go

```diff
@@ -7,6 +7,11 @@ import "github.com/google/uuid"
 type User interface {
 	GetUsername() string
 	GetPassword() string
+	SetPassword(password string) (User, error)
```

Comment:
`SetPassword` should not return User, because we don't create here a new one. User should not return User in any method